src/Spryker/Zed/AppPayment/Business/Payment/Transfer/PaymentTransfer.php

```diff
@@ -0,0 +1,344 @@
+<?php
+
+/**
+ * This file is part of the Spryker Suite.
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace Spryker\Zed\AppPayment\Business\Payment\Transfer;
+
+use ArrayObject;
+use Generated\Shared\Transfer\PaymentsTransmissionsRequestTransfer;
+use Generated\Shared\Transfer\PaymentsTransmissionsResponseTransfer;
+use Generated\Shared\Transfer\PaymentTransfer as GeneratedPaymentTransfer;
+use Generated\Shared\Transfer\PaymentTransmissionTransfer;
+use Spryker\Shared\Log\LoggerTrait;
+use Spryker\Zed\AppPayment\AppPaymentConfig;
+use Spryker\Zed\AppPayment\Business\Exception\PaymentByTenantIdentifierAndOrderReferenceNotFoundException;
+use Spryker\Zed\AppPayment\Business\Message\MessageBuilder;
+use Spryker\Zed\AppPayment\Business\Payment\AppConfig\AppConfigLoader;
+use Spryker\Zed\AppPayment\Dependency\Plugin\PlatformPluginInterface;
+use Spryker\Zed\AppPayment\Persistence\AppPaymentEntityManagerInterface;
+use Spryker\Zed\AppPayment\Persistence\AppPaymentRepositoryInterface;
+use Spryker\Zed\Kernel\Persistence\EntityManager\TransactionTrait;
+use Throwable;
+
+class PaymentTransfer
+{
+    use TransactionTrait;
+    use LoggerTrait;
+
+    /**
+     * @var array<\Generated\Shared\Transfer\PaymentTransmissionTransfer>
+     */
+    protected array $failedPaymentTransmissionTransfers = [];
+
+    /**
+     * @param array<\Spryker\Zed\AppPayment\Dependency\Plugin\PaymentsTransmissionsRequestExtenderPluginInterface> $paymentsTransmissionsRequestExpanderPlugins
+     */
+    public function __construct(
+        protected PlatformPluginInterface $platformPlugin,
+        protected AppPaymentEntityManagerInterface $appPaymentEntityManager,
+        protected AppPaymentRepositoryInterface $appPaymentRepository,
+        protected AppPaymentConfig $appPaymentConfig,
+        protected AppConfigLoader $appConfigLoader,
+        protected array $paymentsTransmissionsRequestExpanderPlugins
+    ) {
+    }
+
+    public function transferPayments(PaymentsTransmissionsRequestTransfer $paymentsTransmissionsRequestTransfer): PaymentsTransmissionsResponseTransfer
+    {
+        // In case all payment transmissions fail, we do not request the platform tzo do something, and we need to return a response with the failed ones.
```

Comment:
```suggestion
        // In case all payment transmissions fail, we do not request the platform to do something, and we need to return a response with the failed ones.
```